### PR TITLE
feat: artifact size metrics

### PR DESCRIPTION
**Pull request by 0marperez**
*Issue #, if available:*
N/A

*Description of changes:*
Added artifact size metrics plugin and misc changes to support the addition.

Note: The plugin currently runs because of the database being bootstrapped in my personal AWS account. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/artifact-size-metrics.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                const hideComment = 
+                  `mutation { 
+                    minimizeComment(input:{subjectId:"${comments[i].id}", classifier:OUTDATED}){
+                      clientMutationId
+                    } 
+                  }`
+            
+                await github.graphql(hideComment)
+              }
+            }
```
Comment (ianbotsf) [inline .github/workflows/artifact-size-metrics.yml:88]: **Suggestion**: Although unlikely, it's possible there could be multiple comments eligible for mutating. This loops over them and synchronously mutates each one but it seems like it'd be more efficient to do that concurrently:

```typescript
const mutations = comments
  .filter(c => c[i].author.login == 'github-actions' && !c[i].isMinimized && c[i].body.startsWith('Affected Artifacts'))
  .map(c =>
    const hideComment = 
      `mutation { 
        minimizeComment(input:{subjectId:"${comments[i].id}", classifier:OUTDATED}){
          clientMutationId
        } 
      }`

    github.graphql(hideComment)
  )
await Promise.all(mutations)
```

**Review by ianbotsf — APPROVED** [review]

File: .github/workflows/artifact-size-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+              }`
+            
+            await github.graphql(writeComment)
+
+      - name: Evaluate
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'acknowledge-artifact-size-increase') }}
+        run: |
+          cd build/reports/metrics
+          cat has-significant-change.txt | grep false || {
+            echo An artifact increased in size by more than allowed or a new artifact was created.
```
Comment (lauzadis) [inline .github/workflows/artifact-size-metrics.yml:107]: suggestion: Add a note about how to work around this error (by adding the `acknowledge-artifact-size-increase` label)

**Review by lauzadis — APPROVED** [review]

File: settings.gradle.kts [hunk 3]
```diff
@@ -4,14 +4,12 @@
...
     repositories {
+        mavenLocal()
         mavenCentral()
         gradlePluginPortal()
         maven {
             name = "kotlinRepoTools"
             url = java.net.URI("https://d2gys1nrxnjnyg.cloudfront.net/releases")
-            content {
-                includeGroupByRegex("""aws\.sdk\.kotlin.*""")
-            }
```
[the remark this answers is not in the document]
Comment (0marperez) [reply settings.gradle.kts:14]: This is so I could import the artifact size metrics plugin. The [plugin ID](https://github.com/awslabs/aws-kotlin-repo-tools/blob/f5605dc44decb02334ff765c7d245a69b8b7699d/build-plugins/build-support/build.gradle.kts#L33) _currently_ doesn't adhere to the regex but I could change it if that makes more sense.

[the remark this answers is not in the document]
Comment (ianbotsf) [reply settings.gradle.kts:14]: Yes, let's fix it up to match our other plugins' conventions and then restore the content filtering for Maven local.

**Review by ianbotsf — APPROVED** [review]

**Pull request merged**